interfaces/builtin/x11.go

```diff
@@ -149,10 +150,26 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+var x11ConnectedPlugMount = []osutil.MountEntry{{
+	Name:    "/var/lib/snapd/hostfs/tmp/.X11-unix/",
+	Dir:     "/tmp/.X11-unix/",
+	Options: []string{"bind"},
+}}
+
 type x11Interface struct {
 	commonInterface
 }
 
+func (iface *x11Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddUpdateNS(`
+	/{,var/lib/snapd/hostfs/}tmp/.X11-unix/ rw,
+	mount options=(rw, bind) /var/lib/snapd/hostfs/tmp/.X11-unix/ -> /tmp/.X11-unix/,
```

Comment:
As above, this will need to be slightly different for non-implicit `x11` slots.

Reply:
Done